test/unit/Transaction.js

```diff
@@ -491,48 +383,67 @@ describe("Transaction", function () {
         });
 
         const signAndAddSignatures = (transaction, ...keys) => {
-            // Map through the keys to sign the transaction and add signatures
-            const signatures = keys.map((key) => {
-                const signature = key.signTransaction(transaction);
-                transaction.addSignature(key.publicKey, signature);
-                return signature;
-            });
+            for (const key of keys) {
+                const sigMap = createSigmap(transaction, key);
+                transaction.addSignature(key.publicKey, sigMap);
```

Comment:
This looks like a changed API for me, I suggest to add an example. Also it seems like it's kinda hard to get the `sigmap` from the transaction.
Would it be possible to create a helper method to do that?
Something like `transaction.addSignature(key.publicKey, transaction.getSigmap());`

Reply:
The `transaction.getSignatures()` returns a `SignatureMap` with all the signatures.
`transaction.addSignature` accepts a public key and even if you pass the whole signature map, only the signatures for the provided public key will be added.

It is not worth creating a helper to get the signatures for a particular public key because that helper will have to iterate through the whole signature map.
The current approach saves one iteration by passing the whole signature map and adding a single public key.
The problem occurs if you want to add signatures for more than one public key.
If you do so, you must iterate n times where n is the number of public keys.

Reply:
Ok, let's add an example to see how it's used from user perspective so I get see it clearly.